src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,137 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private static final int NOT_FOUND_INDEX = -1;
+    private T[] content;
+    private int size;
+
+    public ArrayList() {
+        content = (T[]) new Object[DEFAULT_SIZE];
+    }
+
+    public ArrayList(int customSize) {
+        content = (T[]) new Object[customSize];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == content.length) {
+            grow();
+        }
+        content[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (!(index >= 0 && index <= size)) {
+            throw new ArrayListIndexOutOfBoundsException("The index is out of bounds");
+        }
+        if (size == content.length) {
+            grow();
+        }
+        System.arraycopy(content, index, content, index + 1, size++ - index);
+        content[index] = value;
     }
 
     @Override
     public void addAll(List<T> list) {
+        int listSize = list.size();
+        int currentLength = content.length;
+        if (listSize > currentLength - size) {
+            int missingCells = Math.max(listSize - (currentLength - size), currentLength >> 1);
+            grow(missingCells);
+        }
 
+        T[] listArray = convertListToArray(list);
+        System.arraycopy(listArray, 0, content, size, listSize);
+        size += listSize;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (isInRange(index)) {
+            return content[index];
+        }
+        throw new ArrayListIndexOutOfBoundsException("The index is out of bounds");
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (!isInRange(index)) {
+            throw new ArrayListIndexOutOfBoundsException("Can't set value, because the index "
+                    + "is out of bounds");
```

Comment:
I guess you wanted to print index value in the exception message but you missed it

Reply:
I just wanted to print "BlaBlaBla" to pass all the tests because my brain was melting after doing this homework :D But now I've fixed it.